x-pack/plugins/enterprise_search/server/lib/fetch_indices.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+
+import { ByteSizeValue } from '@kbn/config-schema';
+
+async function fetchIndicesCall(client: IScopedClusterClient): Promise<unknown> {
+  const indexNamesString = 'enterprise-search-index-*';
+
+  // This call retrieves alias and settings (incl. hidden status) information about indices
+  const indices = await client.asCurrentUser.indices.get({
+    index: indexNamesString,
+    expand_wildcards: ['open'],
+    // only get specified index properties from ES to keep the response under 536MB
+    // node.js string length limit: https://github.com/nodejs/node/issues/33960
+    filter_path: [
+      '*.aliases',
+      '*.settings.index.number_of_shards',
+      '*.settings.index.number_of_replicas',
+      '*.settings.index.frozen',
+      '*.settings.index.hidden',
+      '*.data_stream',
+    ],
+    // for better performance only compute aliases and settings of indices but not mappings
+    // @ts-expect-error new param https://github.com/elastic/elasticsearch-specification/issues/1382
+    features: ['aliases', 'settings'],
+  });
+
+  if (!Object.keys(indices).length) {
+    return [];
+  }
+
+  const { indices: indicesStats = {} } = await client.asCurrentUser.indices.stats({
+    index: indexNamesString,
+    expand_wildcards: ['open'],
+    metric: ['docs', 'store'],
+  });
+  const indicesNames = Object.keys(indices);
+  return indicesNames.map((indexName: string) => {
+    const indexData = indices[indexName];
+    const indexStats = indicesStats[indexName];
+    const aliases = Object.keys(indexData.aliases!);
+    return {
+      health: indexStats?.health,
+      status: indexStats?.status,
+      name: indexName,
+      uuid: indexStats?.uuid,
+      primary: indexData.settings?.index?.number_of_shards,
```

Comment:
We could simplify this and not return:
primary
replica
isFrozen
hidden
data_stream

... and if we don't return the `aliases` setting then we can remove the call to fetch settings entirely, I think.

Reply:
Eh... I'm going to amend my last comment. We may want to keep `aliases`. If we decide to let the user create an alias for an index directly in the plugin UI, it will help to show the user what aliases are already defined for a chosen index.

But drop the rest. YAGNI.